mmtrack/datasets/sot_imagenet_vid_dataset.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+from mmdet.datasets import DATASETS
+
+from mmtrack.datasets.parsers import CocoVID
+from .base_sot_dataset import BaseSOTDataset
+
+
+@DATASETS.register_module()
+class SOTImageNetVIDDataset(BaseSOTDataset):
+    """ImageNet VID dataset of single object tracking.
+
+    The dataset only support training mode.
+    """
+
+    def __init__(self, ann_file, *args, **kwargs):
+        """Initialization of SOT dataset class.
+
+        Args:
+            ann_file (str): The coco-format annotation file of ImageNet VID
+                Dataset. It will be loaded and parsed in the
+                `self.load_data_infos` function.
+        """
+        self.coco = CocoVID(ann_file)
+        super().__init__(*args, **kwargs)
+
+    def load_data_infos(self, split='train'):
+        """Load dataset information.
+
+        Args:
+            split (str, optional): The split of dataset. Defaults to 'train'.
+
+        Returns:
+            list[int]: The length of the list is the number of instances. The
+                elemment in the list is instance ID in coco API.
+        """
+        data_infos = list(self.coco.instancesToImgs.keys())
+        return data_infos
+
+    def get_bboxes_from_video(self, video_ind):
+        """Get bbox annotation about the instance in a video. Considering
+        `get_bboxes_from_video` in `SOTBaseDataset` is not compatible with
+        `SOTImageNetVIDDataset`, we oveload this function though it's not
+        called by `self.get_ann_infos_from_video`.
+
+        Args:
+            video_ind (int): video index. Each video_ind denotes an instance.
+
+        Returns:
+            ndarray: in [N, 4] shape. The bbox is in (x, y, w, h) format.
+        """
+        instance_id = self.data_infos[video_ind]
+        img_ids = self.coco.instancesToImgs[instance_id]
+        bboxes = []
+        for img_id in img_ids:
+            for ann in self.coco.imgToAnns[img_id]:
+                if ann['instance_id'] == instance_id:
+                    bboxes.append(ann['bbox'])
+        bboxes = np.array(bboxes).reshape(-1, 4)
+        return bboxes
+
+    def get_img_names_from_video(self, video_ind):
+        """Get all frame paths in a video.
+
+        Args:
+            video_ind (int): video index. Each video_ind denotes an instance.
+
+        Returns:
+            list[str]: all image paths
+        """
+        instance_id = self.data_infos[video_ind]
+        img_ids = self.coco.instancesToImgs[instance_id]
+        img_names = [self.coco.imgs[img_id]['file_name'] for img_id in img_ids]
+        return img_names
+
+    def get_img_infos_from_video(self, video_ind):
```

Comment:
merge `get_img_names_from_video` into `get_img_infos_from_video` in **BaseSOTdataset, GOT10kdataset, TrackingNetDataset, SOTCOCODataset, SOTImageNetVIDDataset**